### PR TITLE
fix(docs): Fix/developer blog

### PR DESCRIPTION
**Pull request by kindavishal**
## Description
This fixes the broken link to developer blog from [gatsby-vendor-partnership](https://www.gatsbyjs.org/docs/gatsby-vendor-partnership/) page. Previously it was trying to point to a directory which didn't exist, now it points to the [blog](https://www.gatsbyjs.org/blog/)

## Related Issues
Fixes 6th task in issue #14693

### CODE REVIEW

File: docs/docs/gatsby-vendor-partnership.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ Depending on your size and reach, there are a few ways we can work together.
 
 1. **Sharing your Gatsby-related content on Twitter.** After you publish your content, and blog about it, just tweet about it and we’re happy to retweet your content to the Gatsby community. We currently (March 2019) have an audience of over 22,000 on Twitter and a typical tweet is seen by around 7,000 developers.
 
-2. **Blogging about your integration.** Our [gatsbyjs.org](gatsbyjs.org) developer blog is open to quality posts from key partners to share about how to use our solutions together. A typical blog post gets 3,000 reads in the first 7 days, and around 10,000 reads over the course of a year.
+2. **Blogging about your integration.** Our [gatsbyjs.org](https://www.gatsbyjs.org/blog/) developer blog is open to quality posts from key partners to share about how to use our solutions together. A typical blog post gets 3,000 reads in the first 7 days, and around 10,000 reads over the course of a year.
```
Comment (m-allanson) [inline docs/docs/gatsby-vendor-partnership.md]:
```suggestion
2. **Blogging about your integration.** Our [gatsbyjs.org](/blog/) developer blog is open to quality posts from key partners to share about how to use our solutions together. A typical blog post gets 3,000 reads in the first 7 days, and around 10,000 reads over the course of a year.
```

File: docs/blog/gatsbygram-case-study/index.md [hunk 2]
```diff
@@ -374,7 +374,7 @@ generates all the configuration for you.
 
 Normally page resources are pre-cached with a service worker. But as several
 browsers (Safari/Microsoft Edge) still don't support Service Workers, the
-[Gatsby `<Link>` component](/packages/gatsby/) pre-caches resources for
+[Gatsby `<Link>` component](https://www.gatsbyjs.org/docs/linking-between-pages/#the-gatsby-link-component) pre-caches resources for
```
Comment (m-allanson) [inline docs/blog/gatsbygram-case-study/index.md]: It's not obvious, but these should be relative links: 

```suggestion
[Gatsby `<Link>` component](/docs/linking-between-pages/#the-gatsby-link-component) pre-caches resources for
```

Comment (kindavishal) [reply docs/blog/gatsbygram-case-study/index.md]: Okkk! I got it!

**Comment by kindavishal** [comment]
> Thanks @kindavishal - one small comment on these is that they should be relative links.

Thanks @m-allanson for informing me this, I was a bit confused between the type of links, your suggestions cleared my doubt, will keep them in mind from next time! :sparkles:

**Pull request merged**